packages/state/src/PositionLeaf.ts

```diff
@@ -86,3 +95,85 @@ function packAsset(asset: PositionAsset) {
     { bytes: 8, value: asset.balance - MIN_INT_64 },
   ])
 }
+
+export async function calculateUpdatedPositions(
```

Comment:
This 100% doesn't belong in this file. Maybe it doesn't even belong in this package, but it definitely needs its own file.